wsmaster/codenvy-hosted-api-audit/src/main/java/com/codenvy/api/audit/server/AuditManager.java

```diff
@@ -0,0 +1,179 @@
+
+            CodenvyLicense license = null;
+            try {
+                license = licenseManager.load();
+            } catch (LicenseException ignored) {
+                //Continue printing report without license info
+            }
+            reportPrinter.printHeader(auditReport, userDao.getTotalCount(), license);
+            printAllUsers(auditReport);
+        } catch (Exception exception) {
+            deleteReportDirectory(auditReport);
+            LOG.error(exception.getMessage(), exception);
+            throw new ServerException(exception.getMessage(), exception);
+        } finally {
+            inProgress.set(false);
+        }
+
+        return auditReport;
+    }
+
+    private void printAllUsers(Path auditReport) throws ServerException {
+        Page<UserImpl> currentPage = userManager.getAll(30, 0);
+        do {
+            //Print users with their workspaces from current page
+            for (UserImpl user : currentPage.getItems()) {
+                List<WorkspaceImpl> workspaces;
+                try {
+                    workspaces = workspaceManager.getWorkspaces(user.getId());
+                } catch (ServerException exception) {
+                    reportPrinter.printError("Failed to receive list of related workspaces for user " + user.getId(), auditReport);
+                    continue;
+                }
+                Map<String, List<AbstractPermissions>> wsPermissions = new HashMap<>();
+                for (WorkspaceImpl workspace : workspaces) {
+                    try {
+                        wsPermissions.put(workspace.getId(), permissionsManager.getByInstance("workspace", workspace.getId()));
+                    } catch (NotFoundException | ConflictException e) {
+                        wsPermissions.put(workspace.getId(), emptyList());
+                    }
+                }
+                reportPrinter.printUserInfoWithHisWorkspacesInfo(auditReport, user, workspaces, wsPermissions);
+            }
+            //Initialize next page if exist, otherwise stop printing report
+            final Optional<PageRef> nextPageRefOpt = currentPage.getNextPageRef();
+            if (nextPageRefOpt.isPresent()) {
+                final PageRef nextPageRef = nextPageRefOpt.get();
+                final long itemsBefore = nextPageRef.getItemsBefore();
+                if (itemsBefore < Integer.MAX_VALUE) {
+                    throw new IllegalArgumentException("Skip count limit was reached while retrieving all users");
+                }
+                currentPage = userManager.getAll(nextPageRef.getPageSize(), (int)itemsBefore);
+            } else {
+                break;
+            }
+        } while (true);
+    }
+
+    @VisibleForTesting
+    void deleteReportDirectory(Path auditReport) {
+        try {
+            //check that deleted directory is created in system temp directory
+            if (auditReport.startsWith(Paths.get(System.getProperty("java.io.tmpdir")))) {
```

Comment:
If somehow the report will be not in temp directory but in root for example, it will delete all parent directory